app/models/concerns/account_statuses_search.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module AccountStatusesSearch
+  extend ActiveSupport::Concern
+
+  def enqueue_update_public_statuses_index
+    if discoverable?
+      enqueue_add_to_public_statuses_index
+    else
+      enqueue_remove_from_public_statuses_index
+    end
+  end
+
+  def enqueue_add_to_public_statuses_index
+    return unless Chewy.enabled?
+
+    AddToPublicStatusesIndexWorker.perform_async(id)
+  end
+
+  def enqueue_remove_from_public_statuses_index
+    return unless Chewy.enabled?
+
+    RemoveFromPublicStatusesIndexWorker.perform_async(id)
+  end
+
+  def add_to_public_statuses_index!
+    return unless Chewy.enabled?
+
+    batch_size = 1000
+    offset = 0
+
+    loop do
```

Comment:
It looks like this is equivalent to `Status.where(account_id: id).find_in_batches(batch_size: 1_000) do |batch|` so the custom loop can be avoided.